crates/flux-middle/src/rty/pretty.rs

```diff
@@ -277,8 +278,25 @@ impl Pretty for Ty {
             }
             TyKind::Param(param_ty) => w!("{}", ^param_ty),
             TyKind::Downcast(adt, .., variant_idx, fields) => {
-                w!("{:?}::{}", adt.did(), ^adt.variant(*variant_idx).name)?;
-                if !fields.is_empty() {
+                let is_struct = adt.is_struct();
```

Comment:
To ponder: maybe `TyKind::Downcast` is something we should hide from users so instead of printing 
```
_1: S { x: i32[a0], y: i32[b1] }
```
we should print 

```
_1.x: i32[a0], _1.y: i32[a1]
```

But perhaps the context that it is an `S` may be relevant to users too

But also `{ x: i32[a0], y: i32[a1] }` doesn't have an equivalence in Rust syntax.

Reply:
Yes, a bit unclear what the right thing here is...